pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BasicAuthBatchIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+  }
+
+  @Test
+  public void testIngestionBatch()
+      throws Exception {
+    File quickstartTmpDir = new File(FileUtils.getTempDirectory(), String.valueOf(System.currentTimeMillis()));
+    FileUtils.forceDeleteOnExit(quickstartTmpDir);
+
+    File baseDir = new File(quickstartTmpDir, "baseballStats");
+    File dataDir = new File(baseDir, "rawdata");
+    File schemaFile = new File(baseDir, SCHEMA_FILE);
+    File configFile = new File(baseDir, CONFIG_FILE);
+    File dataFile = new File(dataDir, DATA_FILE);
+    File jobFile = new File(baseDir, JOB_FILE);
+    Preconditions.checkState(dataDir.mkdirs());
+
+    FileUtils.copyURLToFile(getClass().getResource(BOOTSTRAP_DATA_DIR + "/" + SCHEMA_FILE), schemaFile);
+    FileUtils.copyURLToFile(getClass().getResource(BOOTSTRAP_DATA_DIR + "/" + CONFIG_FILE), configFile);
+    FileUtils.copyURLToFile(getClass().getResource(BOOTSTRAP_DATA_DIR + "/rawdata/" + DATA_FILE), dataFile);
+    FileUtils.copyURLToFile(getClass().getResource(BOOTSTRAP_DATA_DIR + "/" + JOB_FILE), jobFile);
+
+    // patch ingestion job
+    String jobFileContents = IOUtils.toString(new FileInputStream(jobFile));
+    IOUtils.write(jobFileContents.replaceAll("9000", String.valueOf(DEFAULT_CONTROLLER_PORT)),
+        new FileOutputStream(jobFile));
+
+    new BootstrapTableTool("http", "localhost", DEFAULT_CONTROLLER_PORT, baseDir.getAbsolutePath(), AUTH_TOKEN)
+        .execute();
+
+    Thread.sleep(5000);
+
+    // admin with full access
+    JsonNode response = JsonUtils.stringToJsonNode(
+        sendPostRequest("http://localhost:18099/query/sql", "{\"sql\":\"SELECT count(*) FROM baseballStats\"}",
+            AUTH_HEADER));
+    Assert.assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(0).asText(), "LONG",
+        "must return result with LONG value");
+    Assert.assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(0).asText(), "count(*)",
+        "must return column name 'count(*)");
+    Assert.assertEquals(response.get("resultTable").get("rows").get(0).get(0).asInt(), 97889,
+        "must return row count 97889");
+    Assert.assertTrue(response.get("exceptions").isEmpty(), "must not return exception");
+
+    // user with valid auth but no table access
+    JsonNode responseUser = JsonUtils.stringToJsonNode(
+        sendPostRequest("http://localhost:18099/query/sql", "{\"sql\":\"SELECT count(*) FROM baseballStats\"}",
+            AUTH_HEADER_USER));
+    Assert.assertFalse(responseUser.has("resultTable"), "must not return result table");
+    Assert.assertTrue(responseUser.get("exceptions").get(0).get("errorCode").asInt() != 0, "must return error code");
+  }
+
+//  // TODO this endpoint should be protected once UI supports auth
+//  @Test
+//  public void testControllerGetTablesNoAuth()
+//      throws Exception {
+//    System.out.println(sendGetRequest("http://localhost:18998/tables"));
+//  }
```

Comment:
I don't understand the TODO here!

Reply:
we'll wait for the web UI to fully support auth, then protect all the remaining endpoints (including the tables endpoint referenced here)

Reply:
done